### PR TITLE
Fix links in README

### PR DESCRIPTION
**Pull request by nateanl**
Some links couldn't be accessed in the README file (vocab, legacy, etc). Fix them in this PR.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -15,10 +15,10 @@ This repository consists of:
...
 * `examples <https://github.com/pytorch/text/tree/main/examples>`_: Example NLP workflows with PyTorch and torchtext library.
 
-Note: The legacy code discussed in `torchtext v0.7.0 release note <https://github.com/pytorch/text/releases/tag/v0.7.0-rc3>`_ has been retired to `torchtext.legacy <https://github.com/pytorch/text/tree/main/torchtext/legacy>`_ folder. Those legacy code will not be maintained by the development team, and we plan to fully remove them in the future release. See `torchtext.legacy <https://github.com/pytorch/text/tree/main/torchtext/legacy>`_ folder for more details.
+Note: The legacy code discussed in `torchtext v0.7.0 release note <https://github.com/pytorch/text/releases/tag/v0.7.0-rc3>`_ has been retired to `torchtext.legacy <https://github.com/pytorch/text/tree/master/torchtext/legacy>`_ folder. Those legacy code will not be maintained by the development team, and we plan to fully remove them in the future release. See `torchtext.legacy <https://github.com/pytorch/text/tree/master/torchtext/legacy>`_ folder for more details.
```
Comment (parmeet) [inline README.rst:21]: The master branch is not maintained. Also we have removed the legacy folder from main branch (#1422). Perhaps we should add link from the the release branch (https://github.com/pytorch/text/tree/release/0.9) where we first introduced the legacy folder?

File: README.rst [hunk 2]
```diff
@@ -142,9 +142,9 @@ In the v0.9.0 release, we moved the following legacy code to `torchtext.legacy <
 * ``torchtext.legacy.data.pipeline``
 * ``torchtext.legacy.datasets``
 
-We have a `migration tutorial <https://colab.research.google.com/github/pytorch/text/blob/main/examples/legacy_tutorial/migration_tutorial.ipynb>`_ to help users switch to the torchtext datasets in ``v0.9.0`` release. For the users who still want the legacy components, they can add ``legacy`` to the import path.  
+We have a `migration tutorial <https://colab.research.google.com/github/pytorch/text/blob/master/examples/legacy_tutorial/migration_tutorial.ipynb>`_ to help users switch to the torchtext datasets in ``v0.9.0`` release. For the users who still want the legacy components, they can add ``legacy`` to the import path.
```
Comment (parmeet) [inline README.rst:145]: Detto as above.

File: README.rst (hunk 1 above)
Comment (nateanl) [reply README.rst:21]: I agree. Let me change that.

**Review by parmeet — APPROVED** [review]
LGTM! thanks for fixing the links.

**Pull request merged**